langchain/document_loaders/text.ts

```diff
@@ -0,0 +1,40 @@
+import type { readFile as ReadFileT } from 'fs/promises';
+import { Document } from "../docstore/document";
+import { BaseDocumentLoader } from "./base";
+
+let readFile: typeof ReadFileT | null = null;
+
+try {
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  ({ readFile } = require("fs/promises"));
```

Comment:
is `fs` package not a dependency/built in? if not, should we just make it one? how heavy is it - seems like should be light?

Reply:
`fs` is only available in Node.js and not [V8 environments](https://edge-runtime.vercel.app/features/available-apis). Importing `fs` normally prevent users from deploying apps to Cloudflare workers and Vercel Edge functions,  which is a pretty common way to serve these apps. There's no downside to importing `fs` this way as far as I know, though it may be worth looking at this pattern more holistically and deciding if V8 support is a priority.

Reply:
I think supporting deployment to Cloudflare workers is a worthwhile goal, but I'm sure there's other things in the library we'd need to change to make that possible, eg. the "load from hub" functionality

Reply:
There is a way to package TS libs in a manner that allows for different environments. Some features, like reading/writing files on a disk, would be disabled in V8 environments, but these features would work normally in Node.

I can do some digging to see what it would take to implement this at a build/package level.